packages/babel-core/test/fixtures/transformation/misc/regression-4855/expected.js

```diff
@@ -3,4 +3,5 @@
 var _values = values;
 value = _values[fieldName];
 rest = babelHelpers.objectWithoutProperties(_values, [fieldName]);
-var error = void 0;
\ No newline at end of file
+_values;
```

Comment:
It's because `({ [fieldName]: value, ...rest } = values);` is an expression, and we may need to return the result from the expression.